elements/pl-truth-table/pl-truth-table.py

```diff
@@ -0,0 +1,558 @@
+
+    for child in element:
+        if child.tag == "pl-variable":
+            pl.check_attribs(child, [], [])
+            variable_names.append(pl.inner_html(child))
+
+        elif child.tag == "pl-row":
+            pl.check_attribs(child, [], [])
+            rows.append(pl.inner_html(child))
+
+        elif child.tag == "pl-answer-column":
+            pl.check_attribs(child, [], ["expression"])
+            expression = pl.get_string_attrib(child, "expression", None)
+            if expression is None:
+                expression = lxml.html.document_fromstring(
+                        pl.inner_html(child)
+                    ).text_content()
+
+
+            cols.append(AnswerCol(pl.inner_html(child), expression))
+
+        elif child.tag is lxml.etree.Comment:
+            continue
+
+        else:
+            raise ValueError(f"Tags inside of pl-truth-table must be one of pl-variable, pl-row, or pl-answer-column, not '{child.tag}'.")
+
+    return variable_names, rows, cols
+
+
+def prepare(element_html: str, data: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+
+    required_attribs = ["answers-name"]
+    optional_attribs = ["fixed-order", "true-label", "false-label", "num-rows"]
+    pl.check_attribs(element, required_attribs, optional_attribs)
+    name = pl.get_string_attrib(element, "answers-name")
+
+    variable_names, custom_rows, answer_columns = get_question_information(element)
+
+    expressions = [
+        answer_col.expression
+        for answer_col in answer_columns
+    ]
+
+    num_vars = len(variable_names)
+
+    fixed_order = pl.get_boolean_attrib(
+        element, "fixed-order", FIXED_VARIABLES_ORDER_DEFAULT
+    )
+    num_rows = pl.get_integer_attrib(element, "num-rows", None)
+
+    if len(custom_rows) > 0:
+        random.shuffle(custom_rows)
+        fixed_order = True
+    if not fixed_order:
+        random.shuffle(variable_names)
+
+    display_variables = [
+        {"key": str(i), "html": variable}
+        for i, variable in enumerate(variable_names)
+    ]
+
+    display_ans_columns = [
+        {
+            "key": str(i),
+            "html": ans_column.text,
+            "expression": ans_column.expression,
+        }
+        for i, ans_column in enumerate(answer_columns)
+    ]
+
+    display_rows = []
+    correct_answers = []
+
+    #var_vals = {"A": True, "B": False, "C": True, "D": False}
+    all_rows = True
+    row_counter = count(0)
+
+    if all_rows:
+        for value_assignment in product([True, False], repeat=num_vars):
+            var_vals = dict(zip(variable_names, value_assignment))
+
+            display_rows.append({
+                "key": str(next(row_counter)),
+                "values": list(map(str, value_assignment))
+            })
+
+            for expr in expressions:
+                #TODO get rid of this eval call!!!
+                val = eval(expr, var_vals)
+                correct_answers.append(str(val))
+
```

Comment:
For discussion: I was able to substantially simplify the generation logic for variable assignments on each row and the evaluation logic that gets used (should be compatible with a whitelist in sympy).

Before adding more reworked logic here, there seems to be discussion on what row selection options should be available to the question writer as part of this element. I think it seems reasonable to allow custom row selection (as otherwise, this element is functionally capped at 3 variables), but was not sure about what thoughts others have.